vowpalwabbit/core/include/vw/core/example_predict.h

```diff
@@ -42,6 +42,10 @@ class example_predict
   example_predict(example_predict&& other) = default;
   example_predict& operator=(example_predict&& other) = default;
 
+  // this hashing function does not take into account the order of the features
+  // an example with the exact same namespaces / features-values but in a different order will have the same hash
+  uint64_t calculate_order_independent_feature_space_hash();
```

Comment:
```suggestion
  uint64_t get_or_calculate_order_independent_feature_space_hash();
```
Makes the caching of the value clearer